aspnetcore/mvc/models/file-uploads.md

```diff
@@ -1,195 +1,656 @@
+
+For a `files` input element to support uploading multiple files provide the `multiple` attribute on the `<input>` element:
+
+```cshtml
+<input asp-for="FileUpload.FormFiles" type="file" multiple>
 ```
 
-In order to support file uploads, HTML forms must specify an `enctype` of `multipart/form-data`. The `files` input element shown above supports uploading multiple files. Omit the `multiple` attribute on this input element to allow just a single file to be uploaded. The above markup renders in a browser as:
+The individual files uploaded to the server can be accessed through [Model Binding](xref:mvc/models/model-binding) using <xref:Microsoft.AspNetCore.Http.IFormFile>. The sample app demonstrates multiple buffered file uploads for database and physical storage scenarios.
+
+> [!WARNING]
+> Don't rely on or trust the `FileName` property of <xref:Microsoft.AspNetCore.Http.IFormFile> without validation. The `FileName` property should only be used for display purposes and only after HTML encoding the value.
+>
+> The examples provided don't take into account security considerations. See the [Security considerations](#security-considerations) and [Validation](#validation) sections and the sample app for information and examples of file upload validation.
+
+When uploading files using model binding and <xref:Microsoft.AspNetCore.Http.IFormFile>, the action method can accept:
+
+* A single <xref:Microsoft.AspNetCore.Http.IFormFile>.
+* An <xref:System.Collections.IEnumerable>\<<xref:Microsoft.AspNetCore.Http.IFormFile>> or [List](xref:System.Collections.Generic.List`1)\<<xref:Microsoft.AspNetCore.Http.IFormFile>> representing several files. 
 
-![File upload form](file-uploads/_static/upload-form.png)
+The following example:
 
-The individual files uploaded to the server can be accessed through [Model Binding](xref:mvc/models/model-binding) using the [IFormFile](/dotnet/api/microsoft.aspnetcore.http.iformfile) interface. `IFormFile` has the following structure:
+* Loops through one or more uploaded files.
+* Saves the files to the local file system using a file name generated by the app.
+* Returns the total number and size of files uploaded.
 
 ```csharp
-public interface IFormFile
+public async Task<IActionResult> OnPostUploadAsync(List<IFormFile> files)
 {
-    string ContentType { get; }
-    string ContentDisposition { get; }
-    IHeaderDictionary Headers { get; }
-    long Length { get; }
-    string Name { get; }
-    string FileName { get; }
-    Stream OpenReadStream();
-    void CopyTo(Stream target);
-    Task CopyToAsync(Stream target, CancellationToken cancellationToken = null);
+    long size = files.Sum(f => f.Length);
+
+    // Full path to file, including the file name. You can also use
+    // Path.GetRandomFileName to generate a file name without a path
+    // if the app supplies the path (for example, via configuration).
+    var filePath = Path.GetTempFileName();
```

Comment:
Should this say `Path.Combine(MyAppsUploadFileFolder, Path.GetRandomFileName())`? Maybe it will have the user think about a value for for the destination before they copy it in their app?